contracts/verifiers/Verifier.sol

```diff
@@ -752,12 +743,12 @@ abstract contract Verifier is IVerifier, ContextUpgradeable {
         }
     }
 
-    function _checkGroupIdsAndRequestsPerGroup(
-        IVerifier.Request[] calldata requests
-    ) internal view {
+    function _checkGroupIdsAndRequestsPerGroup(IVerifier.Request[] calldata requests) internal {
+        VerifierStorage storage s = _getVerifierStorage();
+
         uint256 newGroupsCount = 0;
         uint256[] memory newGroupsGroupID = new uint256[](requests.length);
-        uint256[] memory newGroupsRequestCount = new uint256[](requests.length);
+        bytes[] memory newGroupsRequestIds = new bytes[](requests.length);
         bool[] memory newGroupsUserIDInRequests = new bool[](requests.length);
```

Comment:
The organisation of calculations and checks looks good enough.
However, I would suggest to use array of structures instead of many arrays.

For example:
```
GroupInfo {
  uint256 id,
  bytes concatenatedRequestIds,
  bool isUserIdInputPresent
}
```

Reply:
I agree. I tried to move it to array of structs but the bytecode exceeded. But I'll try again now with latest updates.